R/TunerIrace.R

```diff
@@ -0,0 +1,94 @@
+#' @title TunerIrace
+#'
+#' @name mlr_tuners_irace
+#' @include Tuner.R
+#'
+#' @description
+#' Subclass for iterated racing tuning calling [irace::irace()] from package \CRANpkg{irace}.
+#'
+#' @section Parameters:
+#' * `show.irace.output` (`logical(1)`)
+#' * `debugLevel` (`integer(1)`)
+#' * `seed` (`integer(1)`)
+#' * `postselection` (`numeric(1)`)
+#' * `elitist` (`integer(1)`)
+#' * `elitistLimit` (`integer(1)`)
+#' * `mu` (`integer(1)`)
+#' * `softRestart` (`integer(1)`)
+#' * `softRestartThreshold` (`numeric(1)`)
+#' * `digits` (`integer(1)`)
+#' * `testType` (`character(1)`)
+#' * `firstTest` (`integer(1)`)
+#' * `eachTest` (`integer(1)`)
+#' * `confidence` (`numeric(1)`)
+#' * `capping` (`integer(1)`)
+#' * `cappingType` (`character(1)`)
+#' * `boundType` (`character(1)`)
+#' * `boundMax` (`numeric(1)`)
+#' * `boundDigits` (`integer(1)`)
+#' * `boundPar` (`numeric(1)`)
+#' * `boundAsTimeout` (`numeric(1)`)
+#'
+#' `show.irace.output` suppresses the output from [irace::irace()] and only prints output from this package.
+#' For the meaning of all other parameters, see [irace::defaultScenario()].
+#' Note that we have removed all control parameters which refer to the termination of the algorithm and
+#' where our terminators allow to obtain the same behavior.
+#'
+#' @templateVar id irace
+#' @template section_dictionary_tuners
+#'
+#' @family Tuner
+#' @export
+#' @examples
+#' # see ?Tuner
+TunerIrace = R6Class("TunerIrace", inherit = Tuner,
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function() {
+      ps = ParamSet$new(list(
+        ParamLgl$new("show.irace.output", default = FALSE),
```

Comment:
Why get rid of "default" though? Just because this goes against the paradox convention of only labelling default if that's in the function? Or do you think user's should be forced to decide?

Reply:
Currently it is "just because" of the convention, but paradox may need to be modified in a way that assumes that users adhere to this convention, e.g. https://github.com/mlr-org/paradox/issues/265